import/WebView.qml

```diff
@@ -0,0 +1,264 @@
+            }
+        }
+    }
+
+    BusyIndicator {
+        id: busySpinner
+        anchors.centerIn: parent
+        running: true
+        visible: webview.loading
+        size: BusyIndicatorSize.Large
+    }
+
+    Timer {
+        id: helper
+        property var geolocationUrls: {{}}
+        property var contextMenu
+
+        function uniqueId() {
+            return Math.floor((Math.random() * 10000000) + 1)
+        }
+
+        function hideVirtualKeyboard() {
+            if (Qt.inputMethod.visible) {
+                webview.focus = true
+            }
+        }
+
+        function openAuthDialog(webView, data, winid) {
+            if (pageStack.busy) {
+                helper.delayedOpenAuthDialog(webView, data, winid)
+            } else {
+                helper.immediateOpenAuthDialog(webView, data, winid)
+            }
+        }
+
+        function immediateOpenAuthDialog(webView, data, winid) {
+            var dialog = pageStack.push(authDialog,
+                                        {"hostname": data.text, "realm": data.title,
+                                         "username": data.storedUsername, "password": data.storedPassword,
+                                         "passwordOnly": data.passwordOnly })
+            dialog.accepted.connect(function () {
+                webView.sendAsyncMessage("authresponse",
+                                         { "winid": winid, "accepted": true,
+                                           "username": dialog.username, "password": dialog.password,
+                                           "dontsave": dialog.dontsave })
+            })
+            dialog.rejected.connect(function() {
+                webView.sendAsyncMessage("authresponse",
+                                         { "winid": winid, "accepted": false})
+            })
+        }
+
+        property var authDialogWebView
+        property var authDialogData
+        property var authDialogWinId
+        function delayedOpenAuthDialog(webView, data, winid) {
+            authDialogWebView = webView
+            authDialogData = data
+            authDialogWinId = winid
+            start()
+        }
+
+        repeat: false
+        running: false
+        interval: 600 // page transition delay.
+        onTriggered: {
+            openAuthDialog(authDialogWebView, authDialogData, authDialogWinId)
+        }
+    }
+
+    Component { id: filePicker; PickerCreator {} }
+    Component { id: multiSelectDialog; MultiSelectDialog {} }
+    Component { id: singleSelectPage; SingleSelectPage {} }
+
+    Component { id: authDialog; AuthDialog {} }
+    Component { id: alertDialog; AlertDialog {} }
+    Component { id: confirmDialog; ConfirmDialog {} }
+    Component { id: locationDialog; LocationDialog {} }
+    Component { id: passwordManagerDialog; PasswordManagerDialog {} }
+    Component { id: promptDialog; PromptDialog {} }
+
+    Component { id: contextMenuComponent; ContextMenu {} }
```

Comment:
Same name as in Silica. I'd not trust that import order always holds. Maybe we should rename this one to something like ```ContextPopup``` so that named imports would not be needed now.